From 3924ff3d7d620f71e6c2cf6c24c0dbbd41f5d259 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Thu, 2 Dec 2021 02:17:19 -0800
Subject: [PATCH] fix time_t fomat error

---
 src/worker-vpn.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/worker-vpn.c b/src/worker-vpn.c
index 37a5b96..aa33ade 100644
--- a/src/worker-vpn.c
+++ b/src/worker-vpn.c
@@ -1619,7 +1619,7 @@ static int tun_mainloop(struct worker_st *ws, struct timespec *tnow)
 	if (WSCONFIG(ws)->switch_to_tcp_timeout &&
 	    DTLS_ACTIVE(ws)->udp_state == UP_ACTIVE &&
 	    tnow->tv_sec > ws->udp_recv_time + WSCONFIG(ws)->switch_to_tcp_timeout) {
-		oclog(ws, LOG_DEBUG, "No UDP data received for %li seconds, using TCP instead\n",
+		oclog(ws, LOG_DEBUG, "No UDP data received for %lld seconds, using TCP instead\n",
 				tnow->tv_sec - ws->udp_recv_time);
 		DTLS_ACTIVE(ws)->udp_state = UP_INACTIVE;
 	}
@@ -2231,7 +2231,7 @@ static int connect_handler(worker_st * ws)
 	} else {
 		time_t expiration = ws->session_start_time + ws->user_config->session_timeout_secs;
 		ret = cstp_printf(ws, "X-CSTP-Session-Timeout: %u\r\n"
-				  "X-CSTP-Session-Timeout-Remaining: %ld\r\n",
+				  "X-CSTP-Session-Timeout-Remaining: %lld\r\n",
 				  ws->user_config->session_timeout_secs,
 				  MAX(expiration - now, 0));
 		SEND_ERR(ret);
-- 
2.17.1

