From d06c686b2bc1278423b2a31399f3bf88e4931126 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Wed, 12 Jan 2022 00:03:21 -0500
Subject: [PATCH] fix ipu

---
 clockdiff.c        | 2 +-
 ping/ping_common.c | 2 +-
 tracepath.c        | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/clockdiff.c b/clockdiff.c
index 538bf78..0d149f9 100644
--- a/clockdiff.c
+++ b/clockdiff.c
@@ -620,7 +620,7 @@ int main(int argc, char **argv)
 				ctl.hisname, ctl.rtt, ctl.rtt_sigma, ctl.min_rtt,
 				ctl.measure_delta, ctl.measure_delta1, s);
 		} else {
-			printf("%ld %d %d\n", now, ctl.measure_delta, ctl.measure_delta1);
+			printf("%lld %d %d\n", now, ctl.measure_delta, ctl.measure_delta1);
 		}
 	}
 	exit(0);
diff --git a/ping/ping_common.c b/ping/ping_common.c
index ba46252..293fa52 100644
--- a/ping/ping_common.c
+++ b/ping/ping_common.c
@@ -883,7 +883,7 @@ int finish(struct ping_rts *rts)
 #endif
 		printf(_(", %g%% packet loss"),
 		       (float)((((long long)(rts->ntransmitted - rts->nreceived)) * 100.0) / rts->ntransmitted));
-		printf(_(", time %ldms"), 1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000);
+		printf(_(", time %lldms"), 1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000);
 	}
 
 	putchar('\n');
diff --git a/tracepath.c b/tracepath.c
index fb4bdca..083cdbb 100644
--- a/tracepath.c
+++ b/tracepath.c
@@ -281,7 +281,7 @@ static int recverr(struct run_state *const ctl)
 		struct timespec res;
 
 		timespecsub(&ts, retts, &res);
-		printf(_("%3ld.%03ldms "), res.tv_sec * 1000 + res.tv_nsec / 1000000,
+		printf(_("%3lld.%03ldms "), res.tv_sec * 1000 + res.tv_nsec / 1000000,
 					   (res.tv_nsec % 1000000) / 1000);
 		if (broken_router)
 			printf(_("(This broken router returned corrupted payload) "));
-- 
2.17.1

