--- a/src/library.c	2021-11-30 22:11:56.071534820 -0800
+++ b/src/library.c	2021-11-30 22:14:32.000000000 -0800
@@ -233,7 +233,7 @@
 
   event_add(cbreg->ev, wait);
 
-  DPRINTF(E_DBG, L_LIB, "Added library callback to %p (id %d), wait %ld.%06ld\n", cbreg->cb, idx_available, wait->tv_sec, wait->tv_usec);
+  DPRINTF(E_DBG, L_LIB, "Added library callback to %p (id %d), wait %lld.%06lld\n", cbreg->cb, idx_available, wait->tv_sec, wait->tv_usec);
 
   return idx_available;
 }
