--- a/src/outputs/alsa.c	2021-11-30 22:21:49.586597904 -0800
+++ b/src/outputs/alsa.c	2021-11-30 22:23:06.000000000 -0800
@@ -900,7 +900,7 @@
   exp_pos = (uint64_t)elapsed * pb->quality.sample_rate / 1000;
   diff = cur_pos - exp_pos;
 
-  DPRINTF(E_SPAM, L_LAUDIO, "counter %d/%d, stamp %lu:%lu, now %lu:%lu, elapsed is %d ms, cur_pos=%" PRIu64 ", exp_pos=%" PRIu64 ", diff=%d\n",
+  DPRINTF(E_SPAM, L_LAUDIO, "counter %d/%d, stamp %llu:%lu, now %llu:%lu, elapsed is %d ms, cur_pos=%" PRIu64 ", exp_pos=%" PRIu64 ", diff=%d\n",
     pb->latency_counter, alsa_latency_history_size, pb->stamp_pts.tv_sec, pb->stamp_pts.tv_nsec / 1000000, ts.tv_sec, ts.tv_nsec / 1000000, elapsed, cur_pos, exp_pos, diff);
 
   // Add the latency to our measurement history
