--- a/nspr/configure	2020-12-28 11:39:44.670418028 +0800
+++ a/nspr/configure	2020-12-28 11:46:38.586100534 +0800
@@ -2334,7 +2334,7 @@
 
 
 ac_aux_dir=
-for ac_dir in ${srcdir}/build/autoconf "$srcdir"/${srcdir}/build/autoconf; do
+for ac_dir in build/autoconf ${srcdir}/build/autoconf; do
   if test -f "$ac_dir/install-sh"; then
     ac_aux_dir=$ac_dir
     ac_install_sh="$ac_aux_dir/install-sh -c"
@@ -2350,7 +2350,7 @@
   fi
 done
 if test -z "$ac_aux_dir"; then
-  as_fn_error $? "cannot find install-sh, install.sh, or shtool in ${srcdir}/build/autoconf \"$srcdir\"/${srcdir}/build/autoconf" "$LINENO" 5
+  as_fn_error $? "cannot find install-sh, install.sh, or shtool in build/autoconf ${srcdir}/build/autoconf" "$LINENO" 5
 fi
 
 # These three variables are undocumented and unsupported,


