--- a/nfnetlink.c	2021-02-23 18:05:22.010667169 +0800
+++ b/nfnetlink.c	2021-02-23 18:06:01.000000000 +0800
@@ -353,16 +353,13 @@
 	if (!nl)
 		return -ENOMEM;
 
-	if (nl_connect(nl, NETLINK_NETFILTER))
+	if (nl_connect_ex(nl, NETLINK_NETFILTER,bufsize,0))
 		return -errno;
 
 	if (nl_socket_add_memberships(nl, NFNLGRP_CONNTRACK_NEW,
 	                                  NFNLGRP_CONNTRACK_DESTROY, 0))
 		return -errno;
 
-	if (nl_socket_set_buffer_size(nl, bufsize, 0))
-		return -errno;
-
 	ufd.cb = handle_event;
 	ufd.fd = nl_socket_get_fd(nl);
 
